tsdoc/src/parser/TokenRange.ts

```diff
@@ -0,0 +1,82 @@
+import { ParserContext } from './ParserContext';
+import { Token } from './Token';
+
+export interface ITokenRangeParameters {
+  parserContext: ParserContext;
+  pos: number;
+  end: number;
+}
+
+/**
+ * Represents a range of tokens extracted from `ParserContext.tokens`.
+ */
+export class TokenRange {
+  /**
+   * The associated parser context that the tokens come from.
+   */
+  public readonly parserContext: ParserContext;
+
+  private _pos: number;
+  private _end: number;
+
+  /**
+   * Constructs a TokenRange object with no tokens.
+   */
+  public static createEmpty(parserContext: ParserContext): TokenRange {
+    return new TokenRange({ parserContext, pos: 0, end: 0 });
+  }
+
+  public constructor(parameters: ITokenRangeParameters) {
+    this.parserContext = parameters.parserContext;
+    this._pos = parameters.pos;
+    this._end = parameters.end;
+    this._validateBounds();
+  }
+
+  /**
+   * The starting index into the associated `ParserContext.tokens` list.
+   */
+  public get pos(): number {
```

Comment:
>pos [](start = 13, length = 3)

Using the same terminology the TS compiler uses to refer to text locations is a little confusing here.

Perhaps call this "tokenStartIndex"/"tokenEndIndex"